### PR TITLE
fix: parse on-ramper api key and more

### PR DESCRIPTION
**Pull request by dennyscode**
Re-Iterating on Ticket: https://lifi.atlassian.net/browse/LF-2786

_Leeroy | Onramper, [12 May 2023, 10:14:59]:
You're appending apiKey with ? change ? > &_

### CODE REVIEW

**Review by chybisov — CHANGES_REQUESTED** [review]

**Review by chybisov — APPROVED** [review]

File: packages/dapp/src/components/OnRamper/OnRamper.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export const OnRamper = () => {
     onRamperConfig.secondaryTextColor
   }&borderRadius=${onRamperConfig.borderRadius}&wgBorderRadius=${
     onRamperConfig.wgBorderRadius
-  }?apiKey=${import.meta.env.VITE_ONRAMPER_API_KEY}`;
+  }&apiKey=${import.meta.env.VITE_ONRAMPER_API_KEY}`;
```
Comment (maxklenk) [inline packages/dapp/src/components/OnRamper/OnRamper.tsx:48]: I would recommend using `URLSearchParams` instead. You can pass it a nice (even typed) object and let it handle the parsing into a query string.  https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#examples

Comment (chybisov) [reply packages/dapp/src/components/OnRamper/OnRamper.tsx:48]: Good suggestion!

Comment (dennyscode) [reply packages/dapp/src/components/OnRamper/OnRamper.tsx:48]: Thanks @maxklenk! Yes, that´s cleaner and I will fix that

File: packages/dapp/src/components/OnRamper/OnRamper.tsx [hunk 2]
```diff
@@ -34,18 +34,25 @@ export const OnRamper = () => {
...
+  searchParams.append('secondaryColor', onRamperConfig.secondaryColor);
+  searchParams.append('cardColor', onRamperConfig.cardColor);
+  searchParams.append('primaryTextColor', onRamperConfig.primaryTextColor);
+  searchParams.append('secondaryTextColor', onRamperConfig.secondaryTextColor);
+  searchParams.append('borderRadius', onRamperConfig.borderRadius.toString());
+  searchParams.append(
+    'wgBorderRadius',
+    onRamperConfig.wgBorderRadius.toString(),
+  );
+  searchParams.append('apiKey', import.meta.env.VITE_ONRAMPER_API_KEY);
```
Comment (chybisov) [inline packages/dapp/src/components/OnRamper/OnRamper.tsx:52]: It should be possible to simplify it to

`const searchParams = new URLSearchParams(onRamperConfig);`

Please check types, maybe conversion of some props to strings will be required 🙂

**Review by chybisov — CHANGES_REQUESTED** [review]

File: packages/dapp/src/components/OnRamper/OnRamper.tsx (hunk 2 above)
Comment (dennyscode) [reply packages/dapp/src/components/OnRamper/OnRamper.tsx]: @chybisov: I have re-iterated on it. What do you think about the latest changes - basically looping through the existing onRamperConfig.

**Review by chybisov — APPROVED** [review]

**Review by Addminus — APPROVED** [review]

**Pull request merged**